### PR TITLE
Fix getCacheKey for DimFilters

### PR DESCRIPTION
**Pull request by b-slim**
getCacheKey was not making separator between two consecutive strings which can lead to a collision of keys

### CODE REVIEW

File: processing/src/main/java/io/druid/query/filter/ExtractionDimFilter.java [hunk 1]
```diff
@@ -74,9 +77,10 @@ public ExtractionFn getExtractionFn()
     byte[] dimensionBytes = StringUtils.toUtf8(dimension);
     byte[] valueBytes = StringUtils.toUtf8(value);
 
-    return ByteBuffer.allocate(1 + dimensionBytes.length + valueBytes.length)
+    return ByteBuffer.allocate(2 + dimensionBytes.length + valueBytes.length)
                      .put(DimFilterCacheHelper.EXTRACTION_CACHE_ID)
                      .put(dimensionBytes)
+                     .put((byte) 0xF)
```
Comment (xvrl) [inline processing/src/main/java/io/druid/query/filter/ExtractionDimFilter.java]: 0xF is a valid control character. All other cache keys use 0xFF as separator which is guaranteed to never show up in any valid UTF-8 stream.

**Comment by b-slim** [comment]
@xvrl thanks ! done

File: processing/src/test/java/io/druid/query/filter/ExtractionDimFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    );
+    ExtractionDimFilter extractionDimFilter2 = new ExtractionDimFilter(
+        "ab",
+        "cd",
+        EasyMock.createMock(ExtractionFn.class),
+        null
+    );
+    Assert.assertFalse(Arrays.equals(extractionDimFilter.getCacheKey(), extractionDimFilter2.getCacheKey()));
+  }
+}
```
Comment (himanshug) [inline processing/src/test/java/io/druid/query/filter/ExtractionDimFilterTest.java]: nit: newline

**Comment by himanshug** [comment]
looks good overall, however , all new files are missing license header

File: processing/src/test/java/io/druid/query/filter/ExtractionDimFilterTest.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package io.druid.query.filter;
```
Comment (fjy) [inline processing/src/test/java/io/druid/query/filter/ExtractionDimFilterTest.java]: header

File: processing/src/test/java/io/druid/query/filter/RegexDimFilterTest.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package io.druid.query.filter;
```
Comment (fjy) [inline processing/src/test/java/io/druid/query/filter/RegexDimFilterTest.java]: header

File: processing/src/test/java/io/druid/query/filter/SearchQueryDimFilterTest.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
+package io.druid.query.filter;
```
Comment (fjy) [inline processing/src/test/java/io/druid/query/filter/SearchQueryDimFilterTest.java]: header

File: processing/src/test/java/io/druid/query/filter/SelectorDimFilterTest.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package io.druid.query.filter;
```
Comment (fjy) [inline processing/src/test/java/io/druid/query/filter/SelectorDimFilterTest.java]: header

**Comment by gianm** [comment]
Looks good to me after headers + squashing

File: processing/src/main/java/io/druid/query/filter/ExtractionDimFilter.java [hunk 7]
```diff
@@ -74,9 +77,10 @@ public ExtractionFn getExtractionFn()
     byte[] dimensionBytes = StringUtils.toUtf8(dimension);
     byte[] valueBytes = StringUtils.toUtf8(value);
 
-    return ByteBuffer.allocate(1 + dimensionBytes.length + valueBytes.length)
+    return ByteBuffer.allocate(2 + dimensionBytes.length + valueBytes.length)
                      .put(DimFilterCacheHelper.EXTRACTION_CACHE_ID)
                      .put(dimensionBytes)
+                     .put((byte) 0xFF)
```
Comment (nishantmonu51) [inline processing/src/main/java/io/druid/query/filter/ExtractionDimFilter.java]: minor nit: can we extract this as a constant in CacheUtil ?

File: processing/src/test/java/io/druid/query/filter/ExtractionDimFilterTest.java (hunk 3 above)
Comment (b-slim) [inline processing/src/test/java/io/druid/query/filter/ExtractionDimFilterTest.java]: done

File: processing/src/test/java/io/druid/query/filter/ExtractionDimFilterTest.java (hunk 2 above)
Comment (b-slim) [inline processing/src/test/java/io/druid/query/filter/ExtractionDimFilterTest.java]: done

File: processing/src/test/java/io/druid/query/filter/RegexDimFilterTest.java (hunk 4 above)
Comment (b-slim) [inline processing/src/test/java/io/druid/query/filter/RegexDimFilterTest.java]: done

File: processing/src/test/java/io/druid/query/filter/SearchQueryDimFilterTest.java (hunk 5 above)
Comment (b-slim) [inline processing/src/test/java/io/druid/query/filter/SearchQueryDimFilterTest.java]: done

File: processing/src/test/java/io/druid/query/filter/SelectorDimFilterTest.java (hunk 6 above)
Comment (b-slim) [inline processing/src/test/java/io/druid/query/filter/SelectorDimFilterTest.java]: done

**Comment by gianm** [comment]
:+1:

**Pull request merged**